components/inputs/input-number.js

```diff
@@ -63,14 +95,18 @@ class InputNumber extends LitElement {
 				class="d2l-input"
 				?disabled="${this.disabled}"
 				id="${this._inputId}"
-				max="${ifDefined(this.max)}"
-				min="${ifDefined(this.min)}"
 				name="${ifDefined(this.name)}"
 				placeholder="${ifDefined(this.placeholder)}"
-				step="${ifDefined(this.step)}"
-				type="number"
-				.value="${this.value}">
+				type="text"
+				.value="${this._formattedValue}">
 		`;
 	}
+
+	async _handleChange(e) {
+		const value = e.target.value;
+		this._formattedValue = value;
+		await this.updateComplete;
+		this.value = parseNumber(value);
```

Comment:
I don't think we should be setting `_formattedValue` here. We should be able to wait for `set value()` to update `_formattedValue` so it doesn't get updated twice.

Reply:
So this is the only way to update the user's input when the input loses focus in certain cases.
For example: Assuming that the input has `max-fraction-digits=2`, the following flow would not work as expected if we don't set `_formattedValue` first then re-set it:
1. Initial value is `1.23`
2. User updates the value to `1.234`
3. The input value will remain `1.234` because no actual change in value was detected because the formatted value for `1.234` would result in `1.23` (same as initial value) due to `max-fraction-digits`

Reply:
Have you looked into why the `input` isn't rerendering the `_formattedValue` correctly? I would have expected the re-render from setting `value` to be enough. Even if you step through it shows the `render` is called with the correct `_formattedValue` but the value inside the `input` isn't correctly getting re-rendered.

Reply:
Yeep, I looked into this quite a bit. The problem is that the `render()` function will be called but the `input` element still detects no change and therefore will not be updated. I expected that setting `value` would be enough as well but I think only the elements with changed properties will be re-rendered.